core/src/test/java/io/jenkins/pluginhealth/scoring/probes/HasUnreleasedProductionChangesProbeTest.java

```diff
@@ -123,5 +124,56 @@ void shouldCheckIfTheUnreleasedCommitsExist() throws IOException, GitAPIExceptio
 
     }
 
+    // check that a commit on pom.xml before the latest release is returning a SUCCESS.
+    @Test
+    void commitOnPomFileBeforeLatestReleaseDateShouldReturnSuccess() throws IOException, GitAPIException {
+        final Path repository = Files.createTempDirectory("test-foo-bar");
+
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+
+        ZonedDateTime releaseTimestamp = ZonedDateTime.now();
+        when(plugin.getReleaseTimestamp()).thenReturn(releaseTimestamp);
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, "")
+        ));
+        when(ctx.getScmRepository()).thenReturn(repository);
+
+        try(Git git = Git.init().setDirectory(repository.toFile()).call()) {
+
+            final Path pom = Files.createFile(repository.resolve("pom.xml"));
+
+            // creating commit
+
+            PersonIdent defaultCommitter = new PersonIdent(git.getRepository());
+            PersonIdent committer = new PersonIdent(defaultCommitter, Date.from(plugin.getReleaseTimestamp().minusDays(1).toInstant()));
+
+            git.add().addFilepattern("pom.xml").call();
+            git.commit().setMessage("Imports pom.xml file").setSign(false).setCommitter(committer).call();
+
+            final HasUnreleasedProductionChangesProbe probe = getSpy();
+
+            for (RevCommit commit : git.log().call()) {
+                long timestamp = commit.getCommitTime();
+                String dateString = plugin.getReleaseTimestamp().toString();
+
+                Instant timestampInstant = Instant.ofEpochSecond(timestamp);
+                DateTimeFormatter formatter = DateTimeFormatter.ISO_ZONED_DATE_TIME;
+                ZonedDateTime zonedDateTime = ZonedDateTime.parse(dateString, formatter);
+                Instant dateInstant = zonedDateTime.toInstant();
+
+                final ProbeResult result = probe.apply(plugin, ctx);
+                assertThat(commit.getFullMessage().equals("Imports pom.xml file"));
+                assert !timestampInstant.isBefore(dateInstant):result.failure("unreleased-production-changes", "unreleased-production-changes does not meet the criteria to be executed on null");
```

Comment:
I need help asserting this.

Reply:
What exactly is the error you're seeing (is this assert statement passing when it shouldn't or vice versa) and what do you expect to see?

Reply:
> What exactly is the error you're seeing (is this assert statement passing when it shouldn't or vice versa) and what do you expect to see?

The issue I am not able to do what I want to do. 

``` assert !timestampInstant.isBefore(dateInstant):result.failure("unreleased-production-changes", "unreleased-production-changes does not meet the criteria to be executed on null");
```

I tried `assertThat` and `assertTrue` but it did not seem to work. 

In the above assert statement where do I call the method to check whether it fails or not?

Reply:
> to check whether it fails or not?

what's wrong with `assert !timestampInstant.isBefore(dateInstant)` or `assertThat(timestampInstant.isBefore(dateInstant)).isEqualTo(true);`?

Reply:
Also, the `assertThat()` call isn't working as you'd expect it to work, it is basically ignoring the results of the assertion. You can change it to:
`assertThat(commit.getFullMessage()).isEqualTo("Imports pom.xml file");` or
`assertThat(commit.getFullMessage().equals("Imports pom.xml file")).isTrue();`

Reply:
>  to check whether it fails or not?
> what's wrong with `assert !timestampInstant.isBefore(dateInstant)` or `assertThat(timestampInstant.isBefore(dateInstant)).isEqualTo(true);`?

Would this syntax test the following-
"a commit on pom.xml before the latest release is returning a SUCCESS"?

Reply:
No, there are some missing pieces.